streams/src/main/java/org/apache/kafka/streams/processor/internals/RecordCollectorImpl.java

```diff
@@ -51,7 +51,10 @@
     public RecordCollectorImpl(final Producer<byte[], byte[]> producer, final String streamTaskId) {
         this.producer = producer;
         offsets = new HashMap<>();
-        logPrefix = String.format("task [%s]", streamTaskId);
+        logPrefix = String.format("task [%s] ", streamTaskId);
+        final LogContext logContext = new LogContext(logPrefix);
```

Comment:
Same here, `LogContext` should be passed in from `StreamTask`

Reply:
Will change it too